test/e2e/rbac/rbac_test.go

```diff
@@ -0,0 +1,143 @@
+package rbac_test
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	"github.com/flightctl/flightctl/test/login"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	nonAdminUser         = "demouser1"
+	adminRoleName        = "rbac-test-admin-role"
+	adminRoleBindingName = "rbac-test-admin-role-binding"
+)
+
+var _ = Describe("RBAC Authorization Tests", Label("rbac", "authorization"), func() {
+	var (
+		harness           *e2e.Harness
+		suiteCtx          context.Context
+		defaultK8sContext string
+		k8sApiEndpoint    string
+	)
+
+	roles := []string{
+		adminRoleName,
+	}
+	roleBindings := []string{
+		adminRoleBindingName,
+	}
+	adminTestLabels := &map[string]string{"test": "rbac-admin"}
+
+	BeforeEach(func() {
+		var err error
+		// Get the harness and context set up by the suite
+		harness = e2e.GetWorkerHarness()
+		suiteCtx = e2e.GetWorkerContext()
+
+		// Get the default K8s context
+		defaultK8sContext, err = harness.GetDefaultK8sContext()
+		Expect(err).ToNot(HaveOccurred(), "Failed to get default K8s context")
+		k8sApiEndpoint, err = harness.GetK8sApiEndpoint(defaultK8sContext)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get Kubernetes API endpoint")
+	})
+
+	AfterEach(func() {
+		harness.ChangeK8sContext(defaultK8sContext)
+		login.LoginToAPIWithToken(harness)
+		harness.CleanupRoles(roles, roleBindings, suiteCtx, flightCtlNs)
+	})
+
+	Context("FlightCtl Admin Role", func() {
+		adminRole := &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      adminRoleName,
+				Namespace: flightCtlNs,
+			},
+		}
+		adminRole.Rules = []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"flightctl.io"},
+				Resources: []string{"*"},
+				Verbs:     []string{"*"},
+			},
+		}
+		adminRoleBinding := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      adminRoleBindingName,
+				Namespace: flightCtlNs,
+			},
+		}
+		adminRoleBinding.RoleRef = rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     adminRoleName,
+		}
+		adminRoleBinding.Subjects = []rbacv1.Subject{
+			{
+				Kind: "User",
+				Name: nonAdminUser,
+			},
+		}
+
+		It("should have full access to all resources and operations", Label("sanity", "83842"), func() {
```

Comment:
I'd make the test a bit more comprehensive by:
1. running the test scenario without role/rolebinding - and expect it to fail with 403
2. running the test again with role/rolebinding created (as you do now)
3. create the role/rolebinding in different NS and expect it to fail with 403
4. create the clusterrole/clusterolebinding and expect it to pass